### DISCUSSION TITLE
fqdn value

### DISCUSSION THREAD

**Discussion by priyakr-git** [projectcontour/contour#4583] [Q&A] [answered]
If you’re running httpproxy in your own cluster (not minikube or kind), what should the value of the fqdn value in the spec.virtualhost section be set to? And do you need a mapping for the fqdn in /etc/hosts?

I am getting a "bad handshake" when I test the websocket over the httpproxy

**Comment by tsaarni** [comment, marked as answer]
The value needs to match with the FQDN that the client uses to reach the proxy. Only the client needs to be able to resolve the name to an IP address. It can be via `/etc/host` at client machine if DNS is not feasible. After establishing TCP connection the client may use the FQDN in two ways: (1) during TLS handshake by sending the server name in TLS SNI extension and (2) during HTTP request by sending `Host` header.  These need to match with the name of the virtualhost.

I guess the reason for the error can be multiple things. Is the problem only happening for websocket or also for simple HTTPS requests?
